bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLogger.java

```diff
@@ -506,78 +498,86 @@ void createNewLog() throws IOException {
     EntryLoggerAllocator getEntryLoggerAllocator() {
         return entryLoggerAllocator;
     }
+
     /**
      * Append the ledger map at the end of the entry log.
      * Updates the entry log file header with the offset and size of the map.
      */
-    private void appendLedgersMap(BufferedLogChannel entryLogChannel) throws IOException {
-        long ledgerMapOffset = entryLogChannel.position();
-
-        ConcurrentLongLongHashMap ledgersMap = entryLogChannel.getLedgersMap();
-        int numberOfLedgers = (int) ledgersMap.size();
+    private void appendLedgersMap(Long ledgerId) throws IOException {
```

Comment:
I think a better change here is to move `appendLedgersMap` as part of `BufferedLogChannel`. so this method doesn't have to be aware of `ledgerId`.  the awareness of `ledgerId` should be in per ledger log manager.

Reply:
yes, it can be moved to BufferedLogChannel